generator/python/build.gradle

```diff
@@ -49,39 +50,41 @@ task generateClient(type: GradleBuild) {
+             versionInNamespace= "v" + versionInNamespace
 
-        packageBase = "${criteoPackage}.${criteoService}.${packageBase}".replace('.', nameSeparator).toString(); // example : criteo_api_marketingsolutions_v2021_10
-        def packageVersion = "${criteoApiVersion}.${generatorVersion}.${getFormattedDate()}"; // example : 2021_10.1.211110
-
-        def generateTask = task("openApiGenerate_python_" + parsedApiName, type: GenerateTask.class) {
+        def namespaceBase = "${criteoPackage}.${criteoService}.${versionInNamespace}".replace('.', nameSeparator).toString(); // example: criteo\api\marketingsolutions\v2021_10
+        
+        def artifactVersion = versionCategory == 'preview' ? '0' : criteoApiVersion.replaceAll("_",".").toString()
+        artifactVersion += ".${generatorVersion}.${getFormattedDate()}".toString() // example : 2021.01.0.211110 for stable , 0.0.211110 for preview
+ 
+        def generateTask = task("openApiGenerate_" + language + "_" + parsedApiName, type: GenerateTask.class) {
                 generatorName = "python"
                 templateDir = "$resourcesDir/templates/".toString()
                 inputSpec = specPath
                 outputDir = clientOutputDirPerVersion
                 removeOperationIdPrefix = true
                 generateApiTests = false
                 generateModelTests = false
-                gitUserId='criteo'
-                gitRepoId='criteo-api-python-sdk'
+                gitUserId = 'criteo'
+                gitRepoId = "criteo-api-${criteoService}-sdk".toString()
                 configOptions = [
-                        apiVersion              : parsedJson.info.version,
-                        packageName             : "${packageBase}".toString(),
-                        invokerPackage          : 'Criteo\\SDK',
-                        packageUrl              : 'https://github.com/criteo/criteo-api-python-sdk',
-                        hideGenerationTimestamp : 'true',
-                        artifactId              : "criteo-api-${criteoService}-sdk-${technologyStack}".toString(),
-                        packageVersion          : "${packageVersion}".toString(), // v2021_10.1.211109
                         groupId                 : 'criteo',
-                        version                 : '1.0.0',
-                        description             : "Python SDK for Criteo API ${criteoService} for ${criteoApiVersion} version".toString(),
-                        sourceCompatibility     : '1.8'
+                        apiVersion              : "${artifactVersion}".toString(),
+                        artifactId              : "criteo-api-${criteoService}-sdk".toString(),
+                        description             : "${language.toUpperCase()} SDK for Criteo API ${criteoService} for ${criteoApiVersion} version".toString(),
+                        hideGenerationTimestamp : 'true',
+                        invokerPackage          : 'Criteo\\SDK',
```

Comment:
shouldn't the invokerPackage be namespaceBase?

Reply:
invokerPackage doesn't seem to have any impact on generated files. Let's keep this potential clean up for another review.